observability-kit-starter/pom.xml

```
@@ -37,8 +42,52 @@
         <dependency>
             <groupId>org.mockito</groupId>
             <artifactId>mockito-inline</artifactId>
-            <version>4.7.0</version>
+            <version>${mockito.version}</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.mockito</groupId>
+            <artifactId>mockito-junit-jupiter</artifactId>
+            <version>${mockito.version}</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.hamcrest</groupId>
+            <artifactId>hamcrest-core</artifactId>
+            <version>2.2</version>
             <scope>test</scope>
         </dependency>
     </dependencies>
+
+    <build>
+        <resources>
+            <resource>
+                <directory>src/main/resources</directory>
+                <filtering>false</filtering>
```

I suggest adding also an exclude rule here, otherwise the `observability-kit.properties` file is copied twice
```suggestion
                <filtering>false</filtering>
                <excludes>
                    <exclude>observability-kit.properties</exclude>
                </excludes>                
```

---

Done! I've also moved the properties file for the agent module in the test resources, so it doesn't get copied from there either.